html/modules/custom/reliefweb_moderation/src/Services/ReportModeration.php

```diff
@@ -221,30 +224,81 @@ public function getFilterDefaultStatuses() {
    * {@inheritdoc}
    */
   public function getEntityFormSubmitButtons($status, EntityModeratedInterface $entity) {
-    $buttons = [
-      'draft' => [
+    $buttons = [];
+    $new = empty($status) || $status === 'draft' || $entity->isNew();
+
+    // Only show save as draft for non-published but editable documents.
+    if ($new || in_array($status, ['draft', 'pending', 'on-hold'])) {
+      $buttons['draft'] = [
         '#value' => $this->t('Save as draft'),
-      ],
-      'to-review' => [
-        '#value' => $this->t('To review'),
-      ],
-      'published' => [
-        '#value' => $this->t('Publish'),
-      ],
-      'on-hold' => [
-        '#value' => $this->t('On-hold'),
-      ],
-      'reference' => [
-        '#value' => $this->t('Reference'),
-      ],
-    ];
+      ];
+    }
 
-    // @todo replace with permission.
-    if (UserHelper::userHasRoles(['administrator', 'webmaster'])) {
-      $buttons['archive'] = [
-        '#value' => $this->t('Archive'),
+    // Editors can publish, put on hold or refuse a document.
+    // @todo use permission.
+    if (UserHelper::userHasRoles(['editor'])) {
+      $buttons = [
+        'draft' => [
+          '#value' => $this->t('Save as draft'),
+        ],
+        'to-review' => [
+          '#value' => $this->t('To review'),
+        ],
+        'published' => [
+          '#value' => $this->t('Publish'),
+        ],
+        'on-hold' => [
+          '#value' => $this->t('On-hold'),
+        ],
+        'reference' => [
+          '#value' => $this->t('Reference'),
+        ],
       ];
     }
+    elseif (UserHelper::userHasRoles(['administrator', 'webmaster'])) {
+      $buttons = [
+        'draft' => [
+          '#value' => $this->t('Save as draft'),
+        ],
+        'to-review' => [
+          '#value' => $this->t('To review'),
+        ],
+        'published' => [
+          '#value' => $this->t('Publish'),
+        ],
+        'on-hold' => [
+          '#value' => $this->t('On-hold'),
+        ],
+        'reference' => [
+          '#value' => $this->t('Reference'),
+        ],
+        'archive' => [
+          '#value' => $this->t('Archive'),
+        ],
+      ];
+    }
+    // Other users can submit for review (or publish directly if trusted).
+    else {
+      $buttons['pending'] = [
```

Comment:
Reports do not have a pending status (it's added in the POST API). This needs to be added to the list of allowed statuses. See https://github.com/UN-OCHA/rwint9-site/pull/723/files#diff-8c045d8e06004cb32ea3cdfdde3e25448dd0ce3fc8873f1e0c77237a54c33061R221 and related code.

BUT the needs for this status should be confirmed with the editorial team.

Reply:
Added (for now)

Reply:
And removed